File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#10 브루트 포스/7568_덩치 (22.04.12 성공).cpp
#include <iostream>
#include <utility>

using namespace std;

int main() {
    int N;
    int cnt = 1;
    pair<int, int> arr[50];
    cin >> N;

    for (int i = 0; i < N; ++i) 
        cin >> arr[i].first >> arr[i].second;
    for (int i = 0; i < N; ++i) {
        cnt = 1;

        for (int j = 0; j < N; j++)
            if (arr[i].first < arr[j].first && arr[i].second < arr[j].second)
                cnt++;
        cout << cnt << " ";
    }

    return 0;
}

File: 백준 풀이/~23.12.31/#10 브루트 포스/7568_덩치 (22.04.12 성공).py
N = int(input())
arr = []

for i in range(N):
    x, y = map(int, input().split())
    arr.append((x, y))

for i in arr:
    cnt = 1
    for j in arr:
        if i[0] < j[0] and i[1] < j[1]:
            cnt += 1
    print(cnt, end=" ")